services/frontend-service/src/ui/components/ServiceLane/EnvSelectionDialog.tsx

```diff
@@ -66,22 +68,35 @@ export const EnvSelectionDialog: React.FC<EnvSelectionDialogProps> = (props) =>
                     : 'Select which environments to run release train to:'
             }
             confirmLabel={props.envSelectionDialog ? 'Remove app from environments' : 'Release Train'}>
-            <div className="envs-dropdown-select">
-                {props.environments.map((env: string, index: number) => {
-                    const enabled = selectedEnvs.includes(env);
-                    return (
-                        <div key={env}>
-                            <Checkbox
-                                enabled={enabled}
-                                onClick={addTeam}
-                                id={String(env)}
-                                label={env}
-                                classes={'env' + env}
-                            />
+            {props.environments.length > 0 ? (
+                <div className="envs-dropdown-select">
+                    {props.environments.map((env: string, index: number) => {
+                        const enabled = selectedEnvs.includes(env);
+                        return (
+                            <div key={env}>
+                                <Checkbox
+                                    enabled={enabled}
+                                    onClick={addTeam}
+                                    id={String(env)}
+                                    label={env}
+                                    classes={'env' + env}
+                                />
+                            </div>
+                        );
+                    })}
+                </div>
+            ) : (
+                <div className="envs-dropdown-select">
+                    {props.envSelectionDialog ? (
+                        <div id="missing_envs">There are no environments to list</div>
+                    ) : (
+                        <div id="missing_envs">
+                            There are no environments that have the selected environment as the upstream target or you
+                            are attempting with an environment group
```

Comment:
Not sure exactly what this means, please clarify in the code.
Maybe split it into 2 sentences.

Reply:
Will make the statement clearer !